Remove workaround in `FreestyleJobTest#should_visit_build_with_permalink`

Starting with https://github.com/jenkinsci/jenkins/pull/8110 by @janfaracik in 2.453, `FreestyleJobTest#should_visit_build_with_permalink` fails with

```
java.lang.AssertionError: 
Build has no changes
Expected: Text matching \QNo changes\E
     but: was "http://127.0.0.1:43213/job/indigenous_warning/lastBuild/"
"1\n2\nDashboard\nindigenous_warning\n#1\nStatus\nChanges\nConsole Output\nEdit Build Information\nDelete build ‘#1’\nKeep this build forever\nStarted 0.8 sec ago\nTook 18 ms\nSuccess\n#1 (May 14, 2024, 3:42:14 PM)\nAdd description\nStarted by anonymous user\nREST API\nJenkins 2.458"
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at core.FreestyleJobTest.should_visit_build_with_permalink(FreestyleJobTest.java:103)
```

This regression is tracked in core as [JENKINS-73168](https://issues.jenkins.io/browse/JENKINS-73168).

This regression destabilized this test suite, as described in #1544. A workaround was added in this repository in #1547. As part of fixing JENKINS-73168, this workaround should be removed.

--- a/src/test/java/core/FreestyleJobTest.java
+++ b/src/test/java/core/FreestyleJobTest.java
@@ -100,8 +100,7 @@ public class FreestyleJobTest extends AbstractJUnitTest {
         b.open();
         assertThat("Permalink link is current URL", driver.getCurrentUrl(), is(expectedUrl));
         assertThat("Build number is correct", b.getNumber(), is(1));
-        // TODO JENKINS-73168
-        //assertThat("Build has no changes", driver, hasContent("No changes"));
+        assertThat("Build has no changes", driver, hasContent("No changes"));
         assertThat("Build is success", b.getResult(), is(Build.Result.SUCCESS.name()));
     }
 

